### PR TITLE
Added JSON import and export capabilities to TargetPlatformModel

### PR DESCRIPTION
**Pull request by lior-dikstein**
Added the 'load_target_platform_model' and 'export_target_platform_model' functions to import and export 'TargetPlatformModel' instances to JSON files with robust validation and error handling. It also includes comprehensive tests for both the 'export_target_platform_model' and 'load_target_platform_model' functions, covering valid use cases, edge cases, and error scenarios.

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/target_platform_capabilities/tpc_io_handler.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+# ==============================================================================
+from pathlib import Path
+from typing import Union
+
+from model_compression_toolkit.logger import Logger
+from model_compression_toolkit.target_platform_capabilities.schema.mct_current_schema import TargetPlatformModel
+import json
+
+
+def load_target_platform_model(tp_model_or_path: Union[TargetPlatformModel, str]) -> TargetPlatformModel:
```
Comment (ofirgo) [inline model_compression_toolkit/target_platform_capabilities/tpc_io_handler.py:23]: change tp model to tpc everywhere in this function (function, variables names, cimments...)
because we know that we are going to change the terminology, but this won't be changed automatically with the refactor to the class name

**Review by ofirgo — APPROVED** [review]
